src/core/thread/mle_router.cpp

```diff
@@ -1309,7 +1310,7 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
     case kDeviceStateChild:
         if ((mDeviceMode & ModeTlv::kModeFFD) && (GetActiveRouterCount() < mRouterUpgradeThreshold))
         {
-            BecomeRouter(ThreadStatusTlv::kTooFewRouters);
+            mRouterSelectionJitterTimer.Start(Timer::SecToMsec(otPlatRandomGet() % kRouterSelectionJitter));
```

Comment:
OK, I think the spec is a bit ambiguous on this.  If the certification tests do not require the jitter as part of the attach process, then I think it's fine to leave this as is.